File: packages/agent-connector/src/adapters/atomcode.js
/**
 * AtomCode adapter — AtomGit GLM OpenAI-compatible chat completions.
 *
 * Reuses LlmDirectAdapter's streaming chat-completions client, but:
 *  - reads ATOMCODE_API_KEY (also accepts LLM_API_KEY / OPENAI_API_KEY)
 *  - reads ATOMCODE_BASE_URL / LLM_BASE_URL / OPENAI_BASE_URL, defaulting to https://api.atomsites.cn/v1
 *  - reads ATOMCODE_MODEL / LLM_MODEL, defaulting to glm-5.1
 *
 * Priority for every value: UI-saved env > process env > default.
 * Stop / status / control flow is inherited from BaseAdapter.
 */

'use strict';

const LlmDirectAdapter = require('./llm-direct');

const DEFAULT_BASE_URL = 'https://api.atomsites.cn/v1';
const DEFAULT_MODEL = 'glm-5.1';

class AtomCodeAdapter extends LlmDirectAdapter {
  constructor(opts) {
    super({
      ...opts,
      adapterLabel: 'AtomCode',
      modelEnvVar: 'ATOMCODE_MODEL',
      suppressConfigLog: true,
    });

    const env = this.agentEnv || process.env;

    const apiKey =
      env.ATOMCODE_API_KEY ||
      env.LLM_API_KEY ||
      env.OPENAI_API_KEY ||
      '';

    const baseUrl = (
      env.ATOMCODE_BASE_URL ||
      env.LLM_BASE_URL ||
      env.OPENAI_BASE_URL ||
      DEFAULT_BASE_URL
    ).replace(/\/$/, '');

    const model =
      env.ATOMCODE_MODEL ||
      env.LLM_MODEL ||
      DEFAULT_MODEL;

    this._apiKey = apiKey;
    this._baseUrl = baseUrl;
    this._model = model;
    this._directMode = !!(this._apiKey && this._baseUrl);

    if (this._directMode) {
      this._log(`AtomCode mode: ${this._baseUrl} model=${this._model}`);
    } else {
      this._log(
        'AtomCode adapter started without API key. ' +
        'Set ATOMCODE_API_KEY via the Launcher Configure screen.'
      );
    }
  }
}

module.exports = AtomCodeAdapter;

File: sdk/src/openagents/adapters/atomcode.py
"""
AtomCode adapter for OpenAgents workspace — AtomGit GLM API.

Mirrors packages/agent-connector/src/adapters/atomcode.js. Reuses the streaming
chat-completions client from NanoClawAdapter but applies AtomCode-specific
defaults (base URL https://api.atomsites.cn/v1, model glm-5.1) and accepts
ATOMCODE_API_KEY in addition to the generic LLM_*/OPENAI_* variables.

Priority for every value: UI-saved env > process env > default.
"""

import logging
import os

from openagents.adapters.nanoclaw import NanoClawAdapter
from openagents.workspace_client import DEFAULT_ENDPOINT

logger = logging.getLogger(__name__)

DEFAULT_BASE_URL = "https://api.atomsites.cn/v1"
DEFAULT_MODEL = "glm-5.1"


class AtomCodeAdapter(NanoClawAdapter):
    """AtomCode (AtomGit GLM) adapter — OpenAI-compatible chat completions."""

    def __init__(
        self,
        workspace_id: str,
        channel_name: str,
        token: str,
        agent_name: str,
        endpoint: str = DEFAULT_ENDPOINT,
        disabled_modules: set | None = None,
        working_dir: str | None = None,
    ):
        super().__init__(
            workspace_id,
            channel_name,
            token,
            agent_name,
            endpoint,
            disabled_modules,
            working_dir,
        )

        # Override direct-mode config with AtomCode-specific env vars
        self._direct_api_key = (
            os.environ.get("ATOMCODE_API_KEY")
            or os.environ.get("LLM_API_KEY")
            or os.environ.get("OPENAI_API_KEY")
            or ""
        )
        self._direct_base_url = (
            os.environ.get("ATOMCODE_BASE_URL")
            or os.environ.get("LLM_BASE_URL")
            or os.environ.get("OPENAI_BASE_URL")
            or DEFAULT_BASE_URL
        ).rstrip("/")
        self._direct_model = (
            os.environ.get("ATOMCODE_MODEL")
            or os.environ.get("LLM_MODEL")
            or DEFAULT_MODEL
        )
        self._direct_mode = bool(self._direct_api_key and self._direct_base_url)

        if self._direct_mode:
            logger.info(
                f"AtomCode mode: {self._direct_base_url} model={self._direct_model}"
            )
        else:
            logger.warning(
                "AtomCode adapter started without API key. "
                "Set ATOMCODE_API_KEY via the Launcher Configure screen."
            )
